vignettes/RPyGeo.Rmd

```diff
@@ -16,143 +21,199 @@ knitr::opts_chunk$set(
+In order to use **RPyGeo** you need a working ArcMap or ArcGIS Pro installation on your computer. 
+In addition, this tutorial requires the following packages to be installed and attached.
 
-```{r, eval = FALSE}
-writeRaster(elev, "C:/workspace/elev.tif")
-st_write(nz, "C:/workspace/nz.shp")
+```{r, eval=FALSE}
+library("RpyGeo")
+library("sf")
+library("raster")
+library("magrittr")
 ```
 
-We write the geographic data`elev` and `nz` to the workspace directory. 
-
-### Load ArcPy module and build environment
-
-First you have to initialize the ArcPy Python site package in R. You can also set environment settings such as `overwrite`, `extension` and `workspace`. It is not necessary to set the `path` parameter if Python is installed in the default location.
-
-The workspace can be a directory or file geodatabase. If you work with large raster files we advise you to use a directory, because it takes a long time to load raster datasets from a file geodatabase into the R session.
-
+**spData** and **RQGIS** must also be installed since we will use the raster object `dem` and the vector object `nz` from these packages to demonstrate both raster and vector operations.
+To make these datasets available for the subsequent ArcMap geoprocessing, we have to save them on disk first.
+Therefore, we export `nz` and `dem` to a temporary directory.
 
 ```{r, eval = FALSE}
-arcpy <- rpygeo_build_env(workspace = "C:/workspace/",
-                        overwrite = TRUE,
-                        extensions = "Spatial")
+data(dem, package = "RQGIS")
+data(nz, package = "spData")
+writeRaster(dem, file.path(tempdir(), "dem.tiff"), format = "GTiff")
+st_write(nz, file.path(tempdir(), "nz.shp"))
 ```
 
-In this example we imported the ArcPy into the R session. It is assigned to the variable `env`. All data is read and written in the `C:/workspace/` directory and files can be overwritten. Moreover, the spatial analysis extension is activated. 
-
-A scratch workspace is automatically created inside the workspace. ArcPy stores temporary files in this directory. You can change the scratch directory with the parameter `scratch_workspace`.
+## Establishing the ArcPy Python interface in R
 
-### Use ArcPy functions
-
-ArcPy functions can be accessed via the `$` operator. You can take advantage of the code completion feature of RStudio. After typing the `$` operator all functions of the ArcPy module are listed. Parameters are also autocompleted. 
+Powered by **reticulate**, `rpygeo_build_env()` will establish an R interface to Python.
+ArcPy will only work with the Python binary that comes with the ArcGIS installation.
+If this Python binary is not installed in the default location (`C:/Python27` in the case of ArcMap and `C:/Program Files/ArcGIS/Pro/bin/Python/envs/arcgispro-py3` in the case of ArcGIS Pro), one has to set it explicitly with the `path` parameter of `rpygeo_build_env()`. 
+Secondly, `rpygeo_build_env()` loads the `arcpy` library.
```

Comment:
Is library the right term here? I would write modules or site-package.

Reply:
Ok, feel free to change back but what is the difference between module, package and site-package?

Reply:
Somewhere in the text it also says that you can extend arcpy with extensions (spatial analyst) and modules (data access)? But what is the difference between extensions and modules. I guess the former must be explicitly activated while the latter is already available (I guess arcpy$da refers to data access...). Would be also great if we could list the available modules and extensions in a table. I remember that you can do it, I read it somewhere in the Zandenberg book. The book, however, is in Jena. Hence, I will look it up next week if you either don't know how to do that.

Sorry, I am not an ArcGIS power user, actually, I almost never use it. Still I am very interested in it! And the good thing is that if I am not aware of the differences stated above, the likelihood is quite high that the same is true for a few other **RPyGeo** users.

Reply:
Yes you are right. They are both Python modules but extensions must to be activated. 

This is very interesting. I was searching for such a function but could not find one. This would be very useful to get a global `rpygeo_search()` function.

A package is a collection of modules with a hierarchy. I think package and site-package are the same thing.